site/app/templates/calendar/Calendar.twig

```diff
@@ -8,10 +8,24 @@
     const isGlobal = {% if not in_course %}true{% else %}false{% endif %};
     const isInstructor = {% if is_instructor %}true{% else %}false{% endif %};
 
-    window.addEventListener("load", function () {
-        loadCalendar({{ view_month }}, {{ view_year }});
+    $( document ).ready(function() {
+        // Get value of the type dropdown
+        var type = $("#calendar-item-type-edit").val();
+        loadCalendar(curr_month, curr_year, curr_day, type);
     });
 
+    window.changeView = function(x){
+        if(x.value === "month"){
+            document.cookie = "view=month";
+        }
+        else if(x.value === "two_week"){
+            document.cookie = "view=two_week";
+        }
+        else if(x.value === "one_week"){
+            document.cookie = "view=one_week";
+        }
```

Comment:
We should try to maintain the eslint formatting, even though eslint can't lint twig files.

```suggestion
        if (x.value === "month") {
            document.cookie = "view=month";
        }
        else if (x.value === "two_week") {
            document.cookie = "view=two_week";
        }
        else if (x.value === "one_week") {
            document.cookie = "view=one_week";
        }
```

Reply:
Thannk you!

Reply:
They're all set with view= before the view type, so in the case we don't have a cookie, I'm pretending there is a view one.